hello/hello_patched.py

```diff
@@ -0,0 +1,113 @@
+import asyncio
+import logging
+from dataclasses import dataclass
+from datetime import timedelta
+
+from temporalio import activity, exceptions, workflow
+from temporalio.client import Client
+from temporalio.worker import Worker
+
+
+# While we could use multiple parameters in the activity, Temporal strongly
+# encourages using a single dataclass instead which can have fields added to it
+# in a backwards-compatible way.
+@dataclass
+class ComposeGreetingInput:
+    greeting: str
+    name: str
+
+
+# Basic activity that logs and does string concatenation
+@activity.defn
+async def compose_greeting(input: ComposeGreetingInput) -> str:
+    activity.logger.info("Running activity with parameter %s" % input)
+    return f"{input.greeting}, {input.name}!"
+
+
+# Basic workflow that execute a activity and fires a timer. Demonstrates how to version a workflow using patched.
+# Steps
+# 1) First run the workflow
+# 2) Ctrl-C and interrupt workflow
+# 3) Comment the current async run function and comment out the async run function with patch (this simulates patching a running workflow)
+# 4) Re-run the workflow, it will detect a workflow is running and pickup the execution following the old code path
+# 5) Re-run the workflow again, since a workflow isn't running it will pickup execution following the new code path
+
+
+@workflow.defn
+class PatchedWorkflow:
+    @workflow.run
+    async def run(self, name: str) -> str:
+        workflow.logger.info("Running workflow with parameter %s" % name)
+        greeting = await workflow.execute_activity(
+            compose_greeting,
+            ComposeGreetingInput("Hello", name),
+            start_to_close_timeout=timedelta(seconds=60),
+        )
+        await asyncio.sleep(120)
+        return greeting
+
+
+#    async def run(self, name: str) -> str:
```

Comment:
Other examples like https://github.com/temporalio/samples-typescript/tree/main/patching-api have all of the workflows as different classes and just swap out as needed. We did this in Python tests too: https://github.com/temporalio/sdk-python/blob/cd8cbf95a898b25ff3f3c34fe78ddacdd65855a1/tests/worker/test_workflow.py#L2003-L2092. That way you don't have to have commented out code, you can just let maybe a CLI arg choose which one is registered.

Reply:
Ok i broke out the files into a patch directory w/readme